qiskit/providers/v2/provider.py

```diff
@@ -0,0 +1,76 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base class for a provider."""
+
+from abc import ABC, abstractmethod
+
+from qiskit.providers.exceptions import QiskitBackendNotFoundError
+
+
+class Provider:
+    """Base common type for all versioned Provider abstract classes.
+
+    Note this class should not be inherited from directly, it is intended
+    to be used for type checking. When implementing a provider you should use
+    the versioned abstract classes as the parent class and not this class
+    directly.
+    """
+    version = 0
+
+
+class ProviderV1(Provider, ABC):
+    """Base class for a Backend Provider."""
+
```

Comment:
This also needs `version = 1`

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5086/commits/0faaee159abf409558a40757ed44136cc65bc0bb